R/update_ard.R

```diff
@@ -0,0 +1,60 @@
+#' Update ARDs
+#'
+#' Functions used to update ARD formatting functions and statistic labels.
+#'
+#' @param x (`data.frame`)\cr
+#'  an ARD data frame of class 'card'
+#' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
```

Comment:
Can you remind me why we aren't allowing them to differentially update groups as well?

Reply:
Ahhh, that is a good point! I was focussing on keeping the API as simple as possible and how I viewed most people using it. But this does limit the function. I am trying to think of an example where I want different formatting for the same statistic, but differently for treatment A vs treatment B 🤔 

But I also would prefer this to meet users' needs if that does come up. What do you think the API for this could look like?

Reply:
We could add an argument to filter generally, e.g. `update_fmt_fn(filter)`, and we could use it like:

```r
update_fmt_fn(filter = group1_level %in% 'GFR')
```

We need to be careful about env handling here, and include tests where GFR is stored in a vector.

Reply:
This is a great addition, thank you! I've added a filter argument. What do you think @bzkrouse ? Is there a better name for the argument?

``` r
library(cards)

ard_continuous(
  ADSL, 
  by = ARM, 
  variables = AGE, 
  statistic = ~ continuous_summary_fns(c("N", "mean", "sd"))
) |>
  update_ard_stat_label(
    stat_names = c("mean", "sd"), 
    stat_label = "Mean (SD)", 
    filter = group1_level == "Placebo"
  )
#> {cards} data frame: 9 x 10
#>   group1 group1_level variable stat_name stat_label   stat
#> 1    ARM      Placebo      AGE         N          N     86
#> 2    ARM      Placebo      AGE      mean  Mean (SD) 75.209
#> 3    ARM      Placebo      AGE        sd  Mean (SD)   8.59
#> 4    ARM    Xanomeli…      AGE         N          N     84
#> 5    ARM    Xanomeli…      AGE      mean       Mean 74.381
#> 6    ARM    Xanomeli…      AGE        sd         SD  7.886
#> 7    ARM    Xanomeli…      AGE         N          N     84
#> 8    ARM    Xanomeli…      AGE      mean       Mean 75.667
#> 9    ARM    Xanomeli…      AGE        sd         SD  8.286
#> ℹ 4 more variables: context, fmt_fn, warning, error
```

<sup>Created on 2024-09-13 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>